libs/wire-api/Readme.md

```diff
@@ -0,0 +1,81 @@
+# Multiverb
+
+We offer the typeclass `AsUnion` to convert a handler return type to a union type
+including all possible responses of a `MultiVerb` endpoint.
+
+Any glue code necessary to convert application types to and from the
+canonical `Union` type corresponding to a `MultiVerb` endpoint should be
+packaged into an `AsUnion` instance.
+
+When using flat sum types, you can use Generics to automatically derive this instance, 
+and for nested types, the following example code should help clarify usage.
+
+```haskell
+data Success = Success
+
+data Failure
+  = InvalidEntry
+  | AccessDenied
+
+-- We need a way to map errors to servant and swagger.
+instance KnownError (MapError e) => IsSwaggerError (e :: Failure) where
+  addToSwagger = addStaticErrorToSwagger @(MapError e)
+
+type instance MapError 'InvalidEntry = 'StaticError 400 "invalid-entry" "Invalid data entered"
+
+type instance MapError 'AccessDenied = 'StaticError 403 "access-denied" "Access denied"
+
+data FailureSuccess = SFFailure Failure | SFSuccess Success
+
+sfToEither :: FailureSuccess -> Either Failure Success
+sfToEither (SFFailure b) = Left b
+sfToEither (SFSuccess d) = Right d
+
+sfFromEither :: Either Failure Success -> FailureSuccess
+sfFromEither (Left b) = SFFailure b
+sfFromEither (Right d) = SFSuccess d
+
+-- type instance ResponseType Failure = Failure
+type instance ResponseType Success = Success
+
+-- ErrorResponse offers facilities to create errors
+type MyErrorResponses =
+  '[ ErrorResponse 'InvalidEntry,
+     ErrorResponse 'AccessDenied
+   ]
+
+-- Responses is a list of errors and a list of success cases.
+type MyResponses =
+  MyErrorResponses .++ '[Success]
+
+accessDenied :: DynError
+accessDenied = dynError @(MapError 'AccessDenied)
+
+invalidEntry :: DynError
+invalidEntry = dynError @(MapError 'InvalidEntry)
+
+failToError :: Failure -> NS I (DynError : DynError : xs)
+failToError = \case
+  -- Z . I wraps the first value in the error list.
+  InvalidEntry -> Z . I $ accessDenied
```

Comment:
I think it would be good to mention tha `Z` and `I` belong to `Data.SOP`. (At least I didn't know about them until @pcapriotti explained them to me.)